docs/development/workflow/development_workflow.rst

```diff
@@ -433,24 +433,42 @@ to GitHub. GitHub will automatically update your pull request.
 Rebase, but only if asked
 =========================
 
-Sometimes the maintainers of Astropy will ask you to *rebase* your changes
-before they are merged into the main Astropy repository.
+Sometimes the maintainers of Astropy may ask a pull request to be *rebased*
+(`git rebase`) or *squashed*/*squished* (`git merge --squash`) before
+changes in a Pull Request are merged into the main Astropy
+*master* repository.
+
+The decisions of when to request a *squash* or *rebase* are left to
+individual maintainers.  These may be requested done to reduce the number of
+visible commits saved in the repositary history, or becauses of code-changes
+in Astropy in the mean-time.  Both involve rewriting the Git history, meaning
+that commit IDs will change, which is why you should do it only if asked.
 
 Conceptually, rebasing means taking your changes and applying them to the latest
 version of the development branch of the official astropy as though that was the
-version you had originally branched from.
-
-Behind the scenes, `git`_ is deleting the changes and branch you made, making the
-changes others made to the development branch of Astropy, then re-making your
-branch from the development branch and applying your changes to your branch.
-This results in re-writing the history of commits, which is why you should do it
-only if asked.
+version you had originally branched from.  Each individual commit remains
+visible, but with new metadata/commit IDs.  A squashed merge changes the
+metadata/commit IDs, and also removes separate visibility of individual commits;
+a new commit and commit message will only contain a textual list of the earlier
+commits.
 
 It is easier to make mistakes rebasing than other areas of `git`_, so before you
 start make a branch to serve as a backup copy of your work::
 
     git branch tmp my-new-feature # make temporary branch--will be deleted later
 
+After altering the history with `git rebase` or `git merge --squash` a normal
+`git push` is prevented, and for both a `git push --force` will be required.
+
+... _howto_rebase:
+
+How to rebase
+=============
+
+Behind the scenes, `git`_ is deleting the changes and branch you made, making the
```

Comment:
@MSeifert04: the "Behind the scenes" part is from the existing documentation, happy to adjust as required.